doc/client_design_doc.md

```diff
@@ -0,0 +1,56 @@
+# SQLFlow Client Design Doc
+
+## Overview
+
+SQLFlow Client implements a minimum client that issues queries to and fetch results/logs from [sqlflowserver](https://github.com/wangkuiyi/sqlflowserver).
+
+In `pysqlflow`, the client will be wrapped in the magic command where user directly writes SQL statements.
+
+## Example
+
+```python
+import sqlflow
+
+client = sqlflow.Client(address='localhost:50051')
```

Comment:
Is base_url or server_url better than address as address is too generic.

Reply:
Done.